services/archiver/archiver.go

```diff
@@ -135,9 +135,11 @@ func doArchive(r *ArchiveRequest) (*models.RepoArchiver, error) {
 	if err == nil {
 		if archiver.Status == models.RepoArchiverGenerating {
 			archiver.Status = models.RepoArchiverReady
-			return archiver, models.UpdateRepoArchiverStatus(ctx, archiver)
+			if err = models.UpdateRepoArchiverStatus(ctx, archiver); err != nil {
+				return nil, err
+			}
 		}
-		return archiver, nil
+		return archiver, commiter.Commit()
```

Comment:
Need when archive already exist in storage, but for some reason the record in db not founded(so it created in tx and all this changes rollbacked here).